tests/integration/runners/test_state.py

```diff
@@ -382,3 +382,88 @@ def test_parallel_orchestrations(self):
             self.assertTrue(received)
             del listener
             signal.alarm(0)
+
+    def test_orchestration_with_pillar_dot_items(self):
+        '''
+        Test to confirm when using a state file that includes other state file, if
+        one of those state files includes pillar related functions that will not
+        be pulling from the pillar cache that all the state files are available and
+        the file_roots has been preserved.  See issues #48277 and #46986.
+        '''
+        self.write_conf({
+            'fileserver_backend': ['roots'],
+            'file_roots': {
+                'base': [self.base_env],
+            },
+        })
+
+        orch_sls = os.path.join(self.base_env, 'main.sls')
+        with salt.utils.fopen(orch_sls, 'w') as fp_:
+            fp_.write(textwrap.dedent('''
+                include:
+                  - one
+                  - two
+                  - three
+            '''))
+
+        orch_sls = os.path.join(self.base_env, 'one.sls')
+        with salt.utils.fopen(orch_sls, 'w') as fp_:
+            fp_.write(textwrap.dedent('''
+                {%- set foo = salt['saltutil.runner']('pillar.show_pillar') %}
+                placeholder_one:
+                  test.succeed_without_changes
+            '''))
+
+        orch_sls = os.path.join(self.base_env, 'two.sls')
+        with salt.utils.fopen(orch_sls, 'w') as fp_:
+            fp_.write(textwrap.dedent('''
+                placeholder_two:
+                  test.succeed_without_changes
+            '''))
+
+        orch_sls = os.path.join(self.base_env, 'three.sls')
+        with salt.utils.fopen(orch_sls, 'w') as fp_:
+            fp_.write(textwrap.dedent('''
+                placeholder_three:
+                  test.succeed_without_changes
+            '''))
+
+        orch_sls = os.path.join(self.base_env, 'main.sls')
+
+        listener = salt.utils.event.get_event(
+            'master',
+            sock_dir=self.master_opts['sock_dir'],
+            transport=self.master_opts['transport'],
+            opts=self.master_opts)
+
+        start_time = time.time()
+        jid = self.run_run_plus(
+            'state.orchestrate',
+            'main',
+            __reload_config=True).get('jid')
+
+        if jid is None:
+            raise Exception('jid missing from run_run_plus output')
```

Comment:
Why do we raise common exceptions, instead of AssertionError?
```python
assert bool(jid), 'Job ID missing from run_run_plus output'
```

Reply:
If you do an assert, it gets removed when the optomized bytecode is compiled at build time.

https://stackoverflow.com/questions/9097706/why-assert-is-not-largely-used

Reply:
Oh, right, my bad. Well, I use so often optimised mode that I totally forgot about that (last 20 years of Python and none of it at all). :wink: 

OK, how about raise other than generic Exception?